blazar/db/sqlalchemy/api.py

```diff
@@ -892,7 +895,7 @@ def host_extra_capability_destroy(host_extra_capability_id):
                 id=host_extra_capability_id,
                 model='ComputeHostExtraCapability')
 
-        session.delete(host_extra_capability[0])
+        host_extra_capability[0].soft_delete(session=session)
```

Comment:
I know it was like this before, but it's always weird to me when a function returns None (or empty), or a list of length exactly one. It makes it pretty easy to assume that it can never return a longer list, when there may not be anything actually enforcing this!